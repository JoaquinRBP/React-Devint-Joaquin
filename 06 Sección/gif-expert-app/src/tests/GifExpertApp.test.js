import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
describe('Pruebas con GifExpertApp', () => {
    test('Deberia cargar el snapshot',()=>{
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar una lista de categorias',()=>{
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper=shallow(<GifExpertApp defaultCategory={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
});