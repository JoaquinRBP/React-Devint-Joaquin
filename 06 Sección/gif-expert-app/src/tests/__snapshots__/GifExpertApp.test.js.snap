// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifExpertApp Deberia cargar el snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Pruebas con GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="0"
    />
    <GifGrid
      category="Dragon Ball"
      key="1"
    />
  </ol>
</div>
`;
